Src/AutoFixture.NUnit3/FixedNameTestMethodBuilder.cs

```diff
@@ -0,0 +1,84 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using System.Globalization;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// Builder that generates <see cref="TestMethod"/> with fixed names.
+    /// This is needed for some test runners such as the Nunit test adaptor for Visual Studio.
+    /// </summary>
+    /// <seealso cref="MutableNameTestMethodBuilder" />
+    public class FixedNameTestMethodBuilder : MutableNameTestMethodBuilder
```

Comment:
Please do not use the inheritance here and implement the interface directly. I see both strategies as independent ones, rather than "FixedName" strategy "is" one "VolatileName" strategy as that is implied by inheritance. Inheritance should be only used when child _is_ parent (like Cat is Animal), but not when two classes look similar and you want to reuse some code.

Another point is that code reuse is really tiny, so inheritance doesn't worse it as maintenance overhead is larger than benefits.

When you refactor this, alter all the protected members in the "VolatileName" strategy as they are no longer required. That will allow to remove most of the guard clauses, as they are not needed for the private methods.